### PR TITLE
[Ethereum] Add simple EIP1559 fee suggestion

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes https://github.com/trustwallet/wallet-core/issues/1605

### CODE REVIEW

File: src/Ethereum/Fee.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include <vector>
+
+using json = nlohmann::json;
+using float128_t = boost::multiprecision::cpp_bin_float_quad;
+using namespace std;
+
+namespace TW::Ethereum::Fee {
+
+auto samplingCurve(double sumWeight) -> double {
+    double sampleMin = 0.1;
```
Comment (catenocrypt) [inline src/Ethereum/Fee.cpp:25]: Should be static local var, or static cost global.

File: src/Ethereum/Fee.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        const auto hex = parse_hex(item[0]);
+        const auto reward = load(hex);
+        if (reward > 0 ) {
+            rewards.push_back(reward);
+        }
+    }
+
+    if (rewards.empty()) {
+        // default 2 gwei
+        return uint256_t(2000000000);
```
Comment (catenocrypt) [inline src/Ethereum/Fee.cpp:52]: Move out to const

File: src/Ethereum/Fee.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    vector<uint256_t> baseFees;
+    vector<size_t> order;
+
+    const auto& array = feeHistory["baseFeePerGas"];
+    for(size_t i = 0; i < array.size(); i++) {
+        const auto hex = parse_hex(array[i]);
+        baseFees.push_back(load(hex));
+        order.push_back(i);
+    }
+    baseFees.back() = baseFees.back() * 9 / 8;
```
Comment (catenocrypt) [inline src/Ethereum/Fee.cpp:92]: Why is this needed?  Maybe add comment

**Review by catenocrypt — APPROVED** [review]
Could you also add C++ test?

**Comment by hewigovens** [comment]
Will do, it failed to compile on Linux

**Review by ahmadSS85 — APPROVED** [review]

**Review by vikmeup — APPROVED** [review]
